src/runtime/offers-flow.js

```diff
@@ -225,19 +231,22 @@ export class OffersFlow {
     return Promise.resolve();
   }
 
+  shouldShow_(clientConfig) {
+    return (
+      !clientConfig.uiPredicates ||
+      clientConfig.uiPredicates.canDisplayAutoPrompt
+    );
+  }
+
   /**
    * Gets the URL that should be used for the activity iFrame view.
-   * @param {!Promise<../model/client-config.ClientConfig>} clientConfigPromise
-   * @return {!Promise<string>}
+   * @param {!../model/client-config.ClientConfig} clientConfig
+   * @return {!string}
```

Comment:
primitive types are non-nullable by default, so this type annotation can be simplified to "@return {string}"